Documentation/Evolution/UnicodeForStringProcessing.md

```diff
@@ -77,7 +77,38 @@ str.contains(/.+e\u{301}/)  // true
 str.contains(/\w+é/)        // true
 ```
 
-Swift's `Regex` follows the level 2 guidelines for Unicode support in regular expressions described in [Unicode Technical Standard #18][uts18], with support for Unicode character classes, canonical equivalence, grapheme cluster matching semantics, and level 2 word boundaries enabled by default. `Regex` provides options for selecting different matching behaviors, such as ASCII character classes or Unicode scalar semantics, which corresponds more closely with other regex engines.
+
+For compatibility with other regex engines and the flexibility to match at both `Character` and Unicode scalar level, you can switch between matching levels for an entire regex or within select portions. This powerful capability provides the expected default behavior when working with strings, while allowing you to drop down for Unicode scalar-specific matching.
+
+By default, literal characters and Unicode scalar values (e.g. `\u{301}`) are coalesced into characters in the same way as a normal string, as shown above. Metacharacters, like `.` and `\w`, and custom character classes each match a single element at the current matching level.
+
+For example, these matches fail, because by the time the parser encounters the "`\u{301}`" Unicode scalar literal, the full `"é"` character has been matched:
+
+```swift
+str.contains(/Caf.\u{301})    // false - `.` matches "é" character
+str.contains(/Caf\w\u{301})   // false - `\w` matches "é" character
+str.contains(/.+\u{301})      // false - `.+` matches each character
+```
+
+Alternatively, we can drop down to use Unicode scalar semantics if we want to match specific Unicode sequences. For example, these regexes matches an `"e"` followed by any modifier with the specified parameters:
+
+```swift
+str.contains(/e[\u{300}-\u{314}]/.matchingSemantics(.unicodeScalar))
+// true - matches an "e" followed by a Unicode scalar in the range U+0300 - U+0314
+str.contains(/e\p{Nonspacing Mark}/.matchingSemantics(.unicodeScalar))
+// true - matches an "e" followed by a Unicode scalar with general category "Nonspacing Mark"
+```
+
+Matching in Unicode scalar mode is analogous to comparing against a string's `UnicodeScalarView` — individual Unicode scalars are matched without combining them into characters or testing for canonical equivalence.
+
+```swift
+str.contains(/Café/.matchingSemantics(.unicodeScalar))
+// false - "e\u{301}" doesn't match with /é/
+str.contains(/Cafe\u{301}/.matchingSemantics(.unicodeScalar))
+// true - "e\u{301}" matches with /e\u{301}/
+```
+
```

Comment:
Somewhere (not sure here or in detailed design) I feel we should have an example using the DSL where we switch modes of a sub-builder. That way we can explain where the implicit grapheme break requirements are inserted (e.g. on entry or on exit?)